src/mscorlib/src/System/ReadOnlySpan.cs

```diff
@@ -0,0 +1,290 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    /// <summary>
+    /// ReadOnlySpan is a uniform API for dealing with arrays and subarrays, strings
+    /// and substrings, and unmanaged memory buffers.  It adds minimal overhead
+    /// to regular accesses and is a struct so that creation and subslicing do
+    /// not require additional allocations.  It is type- and memory-safe.
+    /// </summary>
+    public struct ReadOnlySpan<T>
+    {
+        /// <summary>A byref or a native ptr. Do not access directly</summary>
+        internal /* readonly */ IntPtr _rawPointer;
```

Comment:
It was necessary for the JitHelpers that fetch the field as `ref`. These JitHelpers are not 100% correct anyway wrt GC and we need work in the JIT to fix it - I have removed them and switched the fields to `private readonly`.